mlflow/tracing/fluent.py

```diff
@@ -0,0 +1,131 @@
+import contextlib
+import logging
+from typing import Any, Callable, Dict, Optional
+
+_logger = logging.getLogger(__name__)
+
+
+def trace(
+    _func: Optional[Callable] = None,
+    name: Optional[str] = None,
+    span_type: Optional[str] = None,
+    attributes: Optional[Dict[str, Any]] = None,
```

Comment:
should we expose the setting of user-defined / system-applied metadata and tags here in this signature?

Reply:
Yeah, but one concern I have now is it is quite confusing what should go into attributes/tag/metadata..... Since metadata is not intended to modified by users, maybe we can intentionally not exposing here and use internal setter. WDYT?

Reply:
One tricky thing is tags/metadata are only applicable to the TraceInfo., so adding them in span level API might be confusing. I found we have [mlflow.set_tags](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.set_tags) API for attaching tags to current run, what about doing similar for trace-level tags and metadata?

Reply:
@B-Step62 I agree, we should remove `tags` from spans because spans don't have mutable tags. Attributes should remain, since spans have attributes, right? (cc @BenWilson2)

@B-Step62 Yes, I think it makes sense to extend `mlflow.set_tags()` to operate on traces. Trace-level attributes are a bit trickier...IMO it might be easiest to start by using them for system-defined immutable info & introduce APIs for setting attributes later if needed.

Reply:
Let's collapse all definable elements into attributes. +1 to this